dds/DCPS/ConditionVariable.h

```diff
@@ -59,43 +60,52 @@ class ConditionVariable {
     return CvStatus_Error;
   }
 
-  // Blocks forever if expire_at is zero.
+  /// Block until woken up or until expire_at. Same as wait() if expire_at is zero.
   CvStatus wait_until(const MonotonicTimePoint& expire_at)
   {
-    if (impl_.wait(expire_at.is_zero() ? 0 : &expire_at.value()) == 0) {
+    if (expire_at.is_zero()) {
+      return wait();
+    }
+    if (impl_.wait(&expire_at.value()) == 0) {
       return CvStatus_NoTimeout;
     } else if (errno == ETIME) {
       return CvStatus_Timeout;
     }
     if (DCPS_debug_level) {
-      ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: ConditionVariable::wait_until: %p\n"));
+      ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: ConditionVariable::wait_until: %m\n"));
     }
     return CvStatus_Error;
   }
 
+  /// Block until woken up or for expire_in. Same as wait() if expire_in is zero.
   CvStatus wait_for(const TimeDuration& expire_in)
   {
+    if (expire_in.is_zero()) {
+      return wait();
+    }
```

Comment:
This is the only part that worries me but I don't see anything that is calling this function.  Is that correct?

Reply:
`wait_for`? Yea I don't think anything calls it. It's here because `std::condition_variable` has it, but don't think I ever took the time to see if I could refactor anything to use it. Is the part you're worried about that someone might want 0 to mean to timeout immediately?